mava/wrappers/gym.py

```diff
@@ -0,0 +1,403 @@
+
+        info = {"action_mask": self.get_action_mask(info)}
+        if self.add_global_state:
+            info["global_obs"] = self.get_global_obs(agents_view)
+
+        if self.use_shared_rewards:
+            reward = np.array([np.array(reward).sum()] * self.num_agents)
+        else:
+            reward = np.array(reward)
+
+        return agents_view, reward, terminated, truncated, info
+
+    def get_action_mask(self, info: Dict) -> NDArray:
+        if "action_mask" in info:
+            return np.array(info["action_mask"])
+        return np.ones((self.num_agents, self.num_actions), dtype=np.float32)
+
+    def get_global_obs(self, obs: NDArray) -> NDArray:
+        global_obs = np.concatenate(obs, axis=0)
+        return np.tile(global_obs, (self.num_agents, 1))
+
+
+class SmacWrapper(UoeWrapper):
+    """A wrapper that converts actions step to integers."""
+
+    def step(self, actions: List) -> Tuple[NDArray, NDArray, NDArray, NDArray, Dict]:
+        # Convert actions to integers before passing them to the environment
+        actions = [int(action) for action in actions]
+
+        agents_view, reward, terminated, truncated, info = super().step(actions)
+        info["won_episode"] = info["battle_won"]
+
+        return agents_view, reward, terminated, truncated, info
+
+    def get_action_mask(self, info: Dict) -> NDArray:
+        return np.array(self._env.unwrapped.get_avail_actions())
+
+
+class GymRecordEpisodeMetrics(gymnasium.Wrapper):
+    """Record the episode returns and lengths."""
+
+    def __init__(self, env: gymnasium.Env):
+        super().__init__(env)
+        self._env = env
+        self.running_count_episode_return = 0.0
+        self.running_count_episode_length = 0.0
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[NDArray, Dict]:
+        agents_view, info = self._env.reset(seed, options)
+
+        # Reset the metrics
+        self.running_count_episode_return = 0.0
+        self.running_count_episode_length = 0.0
+
+        # Create the metrics dict
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": False,
+        }
+
+        if "won_episode" in info:
+            metrics["won_episode"] = info["won_episode"]
+
+        info["metrics"] = metrics
+
+        return agents_view, info
+
+    def step(self, actions: NDArray) -> Tuple[NDArray, NDArray, NDArray, NDArray, Dict]:
+        agents_view, reward, terminated, truncated, info = self._env.step(actions)
+
+        self.running_count_episode_return += float(np.mean(reward))
+        self.running_count_episode_length += 1
+
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": np.logical_or(terminated, truncated).all().item(),
```

Comment:
I prefer the `|` feel free to keep `logical_or` though
```suggestion
            "is_terminal_step": (terminated | truncated).all().item(),
```

Reply:
I say we keep the logical_or since it works for regular Python lists unlike the |  operator